core/domain/cleanup_user_subscriptions_model_one_off.py

```diff
@@ -0,0 +1,58 @@
+# coding: utf-8
+#
+# Copyright 2019 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""One-off job for cleaning up UserSubscriptionsModel models in prod."""
+
+from core import jobs
+from core.domain import prod_validation_jobs_one_off
+from core.platform import models
+
+(user_models,) = models.Registry.import_models([models.NAMES.user])
+
+
+class CleanupActivityIdsFromUserSubscriptionsModelOneOffJob(
+        jobs.BaseMapReduceOneOffJobManager):
+    """One off job that removes nonexisting activity ids from
+    UserSubscriptionsModel.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Remove invalid ids in a UserSubscriptionsModel entity."""
+
+        return [user_models.UserSubscriptionsModel]
+
+    @staticmethod
+    def map(model_instance):
+        if not model_instance.deleted:
+            user_subs_model = (
+                prod_validation_jobs_one_off.UserSubscriptionsModelValidator())
+            user_subs_model.fetch_external_models(model_instance)
```

Comment:
Can you just fetch the activity_ids models and nothing else (call datastore_services.fetch_multiple_entities_by_ids_and_models directly)? Otherwise you're fetching lots more than you need to fetch, and it all adds up.

That probably means you can keep _fetch_external_models() private, too.

Reply:
Sounds good!